### PR TITLE
Updates to Tip with unit tests

### PR DESCRIPTION
**Pull request by Jimgskoop**
updates to Tip w/decorators, plus six-way unit tests

### CODE REVIEW

File: scripts/lu-controls/Tip.js [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
  * @constructor
- * @extends Container
- * @requires Loader
- * @version 0.1.3
+ * @extends Abstract
+ * @require Container
+ * @version 0.0.1
  */
-var Container = require( '/scripts/lu-controls/Container' ),
+var Abstract = require( '/scripts/lu-controls/Abstract' ),
```
Comment (Iheartweb) [inline scripts/lu-controls/Tip.js]: Please use `lu/Container` and `lu/Abstract`

Comment (Jimgskoop) [inline scripts/lu-controls/Tip.js]: Do these relative URLs work now?

On Mon, Jun 11, 2012 at 1:55 PM, Robert Martone
reply@reply.github.com
wrote:

> >   */
> > -var Container = require( '/scripts/lu-controls/Container' ),
> > +var Abstract = require( '/scripts/lu-controls/Abstract' ),
> 
> Please use `lu/Container` and `lu/Abstract`
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/linkedin/Lu/pull/153/files#r962793

## 

jimg at yojimg dot net

**Comment by Iheartweb** [comment]
Overall this looks good. Some things to fix:

-There are a lot of path errors in screen.css.
-If the decorators are for tip only they should live under lu-controls/tip.

File: scripts/lu-controls/Tip.js (hunk 1 above)
Comment (Iheartweb) [inline scripts/lu-controls/Tip.js]: with the version of inject I committed on Friday - they should 'work' - fingers crossed.

Comment (Jimgskoop) [inline scripts/lu-controls/Tip.js]: Decorators still use full paths, right?  e.g.

decorators.push("/scripts/lu-decorators/positionAbove");

???

On Mon, Jun 11, 2012 at 4:03 PM, Robert Martone
reply@reply.github.com
wrote:

> >   */
> > -var Container = require( '/scripts/lu-controls/Container' ),
> > +var Abstract = require( '/scripts/lu-controls/Abstract' ),
> 
> with the version of inject I committed on Friday - they should 'work' - fingers crossed.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/linkedin/Lu/pull/153/files#r963723

## 

jimg at yojimg dot net

Comment (Jimgskoop) [inline scripts/lu-controls/Tip.js]: I just did a pull from linkedin/Lu and now I'm seeing network requests
for lu.js at

http://localhost:1337/examples/lu.js

which are 404ing.  This is the same problem I was having on Friday,
and had to revert everything back.  :(

On Mon, Jun 11, 2012 at 4:44 PM, jim g jimg@yojimg.net wrote:

> Decorators still use full paths, right?  e.g.
> 
> decorators.push("/scripts/lu-decorators/positionAbove");
> 
> ???
> 
> On Mon, Jun 11, 2012 at 4:03 PM, Robert Martone
> reply@reply.github.com
> wrote:
> 
> > >   */
> > > -var Container = require( '/scripts/lu-controls/Container' ),
> > > +var Abstract = require( '/scripts/lu-controls/Abstract' ),
> > 
> > with the version of inject I committed on Friday - they should 'work' - fingers crossed.
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/linkedin/Lu/pull/153/files#r963723
> 
> ## 
> 
> jimg at yojimg dot net

## 

jimg at yojimg dot net

Comment (Iheartweb) [inline scripts/lu-controls/Tip.js]: Look at how Lu is required in sandbox.html

**Comment by Jimgskoop** [comment]
-There are a lot of path errors in screen.css.  FIXED
-If the decorators are for tip only they should live under lu-controls/tip.  FIXED

File: config.rb [hunk 2]
```diff
@@ -11,7 +11,8 @@
 # output_style = :expanded or :nested or :compact or :compressed
 
 # To enable relative paths to assets via compass helper functions. Uncomment:
-# relative_assets = true
+# 
+relative_assets = true
```
Comment (Jimgskoop) [inline config.rb:15]: Need some extra eyes on this -- if this is commented out, I get problematic absolute paths in my screen.css.

Comment (Iheartweb) [inline config.rb:15]: I vote that we .gitignore screen.css

Comment (Jimgskoop) [inline config.rb:15]: will do.

On Tue, Jun 12, 2012 at 10:39 PM, Robert Martone
reply@reply.github.com
wrote:

> > @@ -11,7 +11,8 @@
> >  # output_style = :expanded or :nested or :compact or :compressed
> > 
> >  # To enable relative paths to assets via compass helper functions. Uncomment:
> > -# relative_assets = true
> > +#
> > +relative_assets = true
> 
> I vote that we .gitignore screen.css
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/linkedin/Lu/pull/153/files#r973743

## 

jimg at yojimg dot net

**Pull request merged**